crates/jstz_kernel/src/parsing.rs

```diff
@@ -0,0 +1,137 @@
+use jstz_crypto::{hash::Blake2b, public_key_hash::PublicKeyHash};
+use jstz_proto::operation::external::FaDeposit;
+use jstz_proto::{context::account::Address, Result};
+use num_traits::ToPrimitive;
+use tezos_data_encoding::enc::BinWriter;
+use tezos_smart_rollup::michelson::{ticket::FA2_1Ticket, MichelsonContract};
+use tezos_smart_rollup::types::{Contract, PublicKeyHash as TezosPublicKeyHash};
+
+// FIXME: This should return the result instead
+pub fn ticket_hash(ticket: &FA2_1Ticket) -> Blake2b {
+    let mut bytes = Vec::new();
+    ticket.creator().bin_write(&mut bytes).unwrap();
+    ticket.contents().bin_write(&mut bytes).unwrap(); // FIXME: limit how large the content can be
+    Blake2b::from(&bytes)
+}
```

Comment:
I assume the lack of result is due to the anyhow vs error variant issue within the jstz codebase?

We should attach some issue links to these FIXME comments

Reply:
Yeah, I get derive errors when I wrap it as a `Jstz_proto` result variants

Reply:
Btw, removed this func as we now use the TicketHash directly from `FA2_1Ticket`